### PR TITLE
Fix Travis CI configuration

### PR DESCRIPTION
**Pull request by ivuk**
Currently, Travis CI fails to test Python 3.5 and Python 3.6 (on GNU/Linux) properly.

These two commits fix the Python 3.{5,6} tests on GNU/Linux. They also switch the default environment that Travis uses to Trusty, per Travis' recommendations.

Hope you're having a great day! :)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,22 +1,48 @@
...
-os:
-  - linux
-  - osx
-
 matrix:
-  exclude:
-    - os: osx
+  include:
+    - os: linux
+      dist: trusty
```
Comment (costela) [inline .travis.yml]: Wouldn't it be cleaner to set `dist: trusty` on the top-level? Am I missing something?

File: .travis.yml [hunk 2]
```diff
@@ -1,22 +1,48 @@
...
+      sudo: false
```
Comment (costela) [inline .travis.yml]: ...same thing for sudo.

**Review by costela — CHANGES_REQUESTED** [review]
Thanks a lot for spotting this problem! Embarrassing! :sweat_smile: 
Unfortunately I must admit I'm not a big fan of big verbose `.travis.yml` files. Wouldn't it be more concise to use something like this on the top-level?
```
env: TOXENV=py36,py35,py34,py27
python:
  - '3.6'
  - '3.5'
  - '3.4'
  - '2.7'
matrix:
  include:
    - env: TOXENV=flake8
```
Not sure this works as I imagine, but we can give it a try.

Maybe you could even go ahead and add
```
matrix:
  fast_finish: true
```
OSX tests are totally broken anyway... :wink:

File: .travis.yml [hunk 3]
```diff
@@ -2,21 +2,34 @@ language: python
...
+  - '3.4'
+  - '2.7'
 
 os:
   - linux
   - osx
 
 matrix:
+  include:
+    - env: TOXENV=py27
```
Comment (costela) [inline .travis.yml]: Maybe we can try setting `TOXENV=py` globally? That could make it use only the default python (avoiding the duplicate tests we were getting with the previous versions). Not sure, but worth a try :wink:

**Review by costela — CHANGES_REQUESTED** [review]

**Comment by ivuk** [comment]
I don't really see a nice way to avoid unrolling the build matrix at this point. It seems other projects, such as `pip` or `simplejson`, also use the complete matrix specified by hand. I guess it's the easiest way to make Travis get the proper environment variables for a specific interpreter version.

I'll try to work around this a bit more, but at this time I feel that the most we can do is abstract the `dist` and `sudo` variables from the matrix itself.

**Comment by costela** [comment]
I have one last idea. Give me a second.

**Comment by costela** [comment]
OK, I think I got it. Before I merge, maybe you'd like to squash a few of your test commits? (you'll probably have to pull, since I pushed to your branched)
Thanks again for the help! :smiley: :+1:

**Comment by ivuk** [comment]
I guess this should be OK now. :)

**Review by costela — APPROVED** [review]

**Comment by costela** [comment]
Perfect, thanks! :smiley: :+1:

**Pull request merged**